helix-core/src/test/java/org/apache/helix/cloud/virtualTopologyGroup/AssignmentEvaluation.java

```diff
@@ -0,0 +1,126 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+public class AssignmentEvaluation {
+  private final Map<String, Set<String>> _assignment;
+  private final Map<String, Set<String>> _coveredZonesByVirtualGroup;
+  private final Map<String, Set<String>> _affectedVirtualGroupByZone;
+  private final Map<String, String> _zoneByInstance;
+  private final Map<String, String> _virtualGroupByInstance;
+
+  public AssignmentEvaluation(Map<String, Set<String>> assignment, Map<String, Set<String>> zoneMapping) {
+    _assignment = assignment;
+
+    _zoneByInstance = new HashMap<>();
+    _virtualGroupByInstance = new HashMap<>();
+    for (Map.Entry<String, Set<String>> entry : zoneMapping.entrySet()) {
+      entry.getValue().forEach(instance -> _zoneByInstance.put(instance, entry.getKey()));
+    }
+    // key is virtual group, value is the set of physical zones that the virtual group span across
+    _coveredZonesByVirtualGroup = new HashMap<>();
+    // key is physical zone, value is the set of affected virtual group if the physical zone is down
+    _affectedVirtualGroupByZone = new HashMap<>();
+    for (String virtualGroup : assignment.keySet()) {
+      _coveredZonesByVirtualGroup.put(virtualGroup, computeCoveredZones(_zoneByInstance, assignment.get(virtualGroup)));
+      for (String instance : assignment.get(virtualGroup)) {
+        String zone = _zoneByInstance.get(instance);
+        _virtualGroupByInstance.put(instance, virtualGroup);
+        _affectedVirtualGroupByZone.putIfAbsent(zone, new HashSet<>());
+        _affectedVirtualGroupByZone.get(zone).add(virtualGroup);
+      }
+    }
+  }
+
+  public void print() {
+    for (Map.Entry<String, Set<String>> entry : _coveredZonesByVirtualGroup.entrySet()) {
+      System.out.println("virtual group: " + entry.getKey() + ", " + "covered physical zones: " + entry.getValue());
+    }
+    for (Map.Entry<String, Set<String>> entry : _affectedVirtualGroupByZone.entrySet()) {
+      System.out.println("physical zone " + entry.getKey() + " impacts " + entry.getValue());
+    }
+    System.out.println(generateStats().toString());
```

Comment:
Are all these prints useful? If it won't help for any debugging purpose, but only gets used in one time testing. I suggest to remove them.